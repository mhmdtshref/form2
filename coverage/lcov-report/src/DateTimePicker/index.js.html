<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/DateTimePicker/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/DateTimePicker</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">69.23% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/13</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
&nbsp;
class DateTimePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: this.props.value || <span class="branch-1 cbranch-no" title="branch not covered" >this.dateToDateTime(new Date()),</span>
    };
  }
&nbsp;
  onChange = <span class="fstat-no" title="function not covered" >({</span> target }) =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ value: target.value }, <span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      this.props.formstate(this.props.name, this.state.value);</span>
<span class="cstat-no" title="statement not covered" >      if(this.props.onChange) <span class="cstat-no" title="statement not covered" >this.props.onChange();</span></span>
    });
  };
&nbsp;
  generateProps = (props) =&gt; {
    let newProps = {...props};
    delete newProps.onChange;
    delete newProps.formstate;
    delete newProps.value;
    return newProps;
  };
&nbsp;
  dateToDateTime = <span class="fstat-no" title="function not covered" >(d</span>ate) =&gt; {
<span class="cstat-no" title="statement not covered" >      return date.toISOString().slice(0,16);</span>
  };
&nbsp;
&nbsp;
  componentDidMount() {
      this.props.formstate(this.props.name, this.props.value || <span class="branch-1 cbranch-no" title="branch not covered" >this.dateToDateTime(new Date()))</span>;
  }
&nbsp;
    render() {
    return &lt;input type='datetime-local' value={this.state.value} onChange={this.onChange} {...this.generateProps(this.props)} /&gt;;
  }
}
&nbsp;
export default DateTimePicker;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jul 09 2019 12:33:04 GMT+0300 (Eastern European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
