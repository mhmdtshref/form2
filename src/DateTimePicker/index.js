import React, { Component } from 'react'

class DateTimePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: (new Date()).toISOString(),
    };
  }

  onChange = ({ target }) => {
    this.setState({ value: target.value }, () => {
      this.props.formstate(this.props.name, this.state.value);
      if(this.props.onChange) this.props.onChange();
    });
  };

  generateProps = (props) => {
    let newProps = {...props};
    delete newProps.onChange;
    delete newProps.formstate;
    delete newProps.value;
    return newProps;
  };

    render() {
    return <input type='datetime-local' value={this.state.value} onChange={this.onChange} {...this.generateProps(this.props)} />;
  }
}

export default DateTimePicker;
