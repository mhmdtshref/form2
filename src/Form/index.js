import React, { Component } from 'react'

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      values: {}
    }
  }

  setStateValue = (name, value) => {
    this.setState((prevState) => ({...prevState.values, [name]: value}), () => {
      console.log("STATE CHANGED:::", this.state);
    })
  };

  setEventToState = ({ target }) => {
    this.setStateValue([target.name], target.value)
  };

  renderTextField = (field, index) => {
    field.key = field.key ? field.key : index;
    if (field.onChange) {
      return <input
        type='text'
        name={field.name}
        onChange={(event) => { field.onChange(event); this.setEventToState(event) }}
        {...field}
      />
    } else {
      return <input
        type='text'
        name={field.name}
        onChange={this.setEventToState}
        {...field}
      />
    }
  };

  renderFields = (fields) => {
    return fields.map((field, index) => {
      switch (field.type) {
        case 'text':
          return this.renderTextField(field, index)
      }
    })
  };


  renderChildren = () => {
    const { setStateValue } = this;
    return React.Children.toArray(this.props.children).map((child, key) => React.cloneElement(child, { formstate: setStateValue, key }));
  };

  render() {
    return <form>
      { this.renderChildren() }
    </form>
  }
}

export default Form
