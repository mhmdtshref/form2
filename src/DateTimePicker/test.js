import React from 'react';
import { render, mount } from 'enzyme';
import jestMock from 'jest-mock';
import DateTimePicker from "./index";
import Form from "../Form";

describe('Password Field Test', () => {

  it('is truthy', () => {
    expect(DateTimePicker).toBeTruthy()
  });


  it('Test natural date picker field', () => {
    const dateTimePicker = render(
        <Form>
            <DateTimePicker name="birthday" value="1996-07-26" />
        </Form>
    );
    expect(dateTimePicker).toMatchSnapshot();
  });


  it('Test date picker field with default value property', () => {
    const dateTimePicker = render(
        <Form>
            <DateTimePicker name="birthday" value="1996-07-26" />
        </Form>
    );
    expect(dateTimePicker).toMatchSnapshot();
  });


  it('Test date picker field onChange', () => {
    const event = {
        target: {
            value: '2019-07-05'
        }
    };

    const dateTimePickerComponent = mount(
        <Form>
            <DateTimePicker name="birthday" value="1996-07-26" />
        </Form>
    );

    const dateTimePicker = dateTimePickerComponent.find('input').first();
    dateTimePicker.simulate('change', event);
  });


  it('Test date picker field with on onChange in properties', () => {
        const onChangeHandler = jestMock.fn();
        const event = {
            target: {
                value: '2019-07-05'
            }
        };

        const dateTimePickerComponent = mount(
            <Form>
                <DateTimePicker name="birthday" onChange={onChangeHandler} />
            </Form>
        );

        const dateTimePicker = dateTimePickerComponent.find('input').first();
        dateTimePicker.simulate('change', event);

        expect(onChangeHandler).toBeCalled();
    });

});
