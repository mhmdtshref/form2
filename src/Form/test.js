import React from 'react';
import renderer from 'react-test-renderer';
import Form from './index';
import TextField from "../TextField";
import PasswordField from "../PasswordField";
import DatePicker from '../DatePicker';

describe('Form Test', () => {

  it('is truthy', () => {
    expect(Form).toBeTruthy()
  });

  it('Form with one field in it', () => {
    const form = renderer.create(<Form>
      <TextField name="username" />
    </Form>).toJSON();
    expect(form).toMatchSnapshot();
  });

  it('Form with multiple fields in it', () => {
    const form = renderer.create(<Form>
      <TextField name="username" placeholder="Username"/>
      <PasswordField name="password" placeholder="email" />
      <DatePicker name="birthday" />
    </Form>).toJSON();
    expect(form).toMatchSnapshot();
  });
});
