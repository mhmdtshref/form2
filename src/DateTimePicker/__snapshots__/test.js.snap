// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Field Test Test date picker field with default value property 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "name": "birthday",
          "type": "datetime-local",
          "value": "1996-07-26",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Password Field Test Test natural date picker field 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "name": "birthday",
          "type": "datetime-local",
          "value": "1996-07-26",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
